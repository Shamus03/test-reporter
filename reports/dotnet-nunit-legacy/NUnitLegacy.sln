
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio 15
VisualStudioVersion = 15.0.28307.1927
MinimumVisualStudioVersion = 10.0.40219.1
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "NUnitLegacyTests", "NUnitLegacyTests\NUnitLegacyTests.csproj", "{28C713F2-6E53-41A0-9B2C-6AC6FF1DB80B}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MyLibrary", "MyLibrary\MyLibrary.csproj", "{157F3352-FB50-4C32-A555-8FF98888CCB6}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{28C713F2-6E53-41A0-9B2C-6AC6FF1DB80B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{28C713F2-6E53-41A0-9B2C-6AC6FF1DB80B}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{28C713F2-6E53-41A0-9B2C-6AC6FF1DB80B}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{28C713F2-6E53-41A0-9B2C-6AC6FF1DB80B}.Release|Any CPU.Build.0 = Release|Any CPU
		{157F3352-FB50-4C32-A555-8FF98888CCB6}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{157F3352-FB50-4C32-A555-8FF98888CCB6}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{157F3352-FB50-4C32-A555-8FF98888CCB6}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{157F3352-FB50-4C32-A555-8FF98888CCB6}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {D64049EC-5C5F-4FBE-AC6C-2DBB54DD3D6F}
	EndGlobalSection
EndGlobal
