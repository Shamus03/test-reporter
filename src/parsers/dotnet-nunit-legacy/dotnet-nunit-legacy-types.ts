export interface NunitReport {
  'test-results': TestResults
}

export interface TestResults {
  $: {
    time: string
  }
  'test-suite'?: TestSuite[]
}

export interface TestSuite {
  $: {
    type: string
    name: string
  }
  results: TestSuiteResult[]
}

export interface TestSuiteResult {
  'test-case'?: TestCase[]
  'test-suite'?: TestSuite[]
}

export interface TestCase {
  $: {
    name: string
    result: string
    time: string
  }
  failure?: TestFailure[]
  reason?: TestFailure[]
}

export interface TestFailure {
  message?: string[]
  'stack-trace'?: string[]
}
