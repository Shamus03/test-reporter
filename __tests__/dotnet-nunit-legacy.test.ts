import * as fs from 'fs'
import * as path from 'path'

import {DotnetNunitLegacyParser} from '../src/parsers/dotnet-nunit-legacy/dotnet-nunit-legacy-parser'
import {ParseOptions} from '../src/test-parser'
import {getReport} from '../src/report/get-report'
import {normalizeFilePath} from '../src/utils/path-utils'

describe('dotnet-nunit-legacy tests', () => {
  it('report from ./reports/dotnet test results matches snapshot', async () => {
    const fixturePath = path.join(__dirname, 'fixtures', 'dotnet-nunit-legacy.xml')
    const outputPath = path.join(__dirname, '__outputs__', 'dotnet-nunit-legacy.md')
    const filePath = normalizeFilePath(path.relative(__dirname, fixturePath))
    const fileContent = fs.readFileSync(fixturePath, {encoding: 'utf8'})

    const opts: ParseOptions = {
      parseErrors: true,
      trackedFiles: ['MyLibrary/Calculator.cs', 'NUnitLegacyTests/CalculatorTests.cs']
    }

    const parser = new DotnetNunitLegacyParser(opts)
    const result = await parser.parse(filePath, fileContent)
    expect(result).toMatchSnapshot()

    const report = getReport([result])
    fs.mkdirSync(path.dirname(outputPath), {recursive: true})
    fs.writeFileSync(outputPath, report)
  })
})
