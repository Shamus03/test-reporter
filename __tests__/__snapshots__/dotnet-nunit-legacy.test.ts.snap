// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotnet-nunit-legacy tests report from ./reports/dotnet test results matches snapshot 1`] = `
TestRunResult {
  "path": "fixtures/dotnet-nunit-legacy.xml",
  "suites": [
    TestSuiteResult {
      "groups": [
        TestGroupResult {
          "name": "CalculatorTests",
          "tests": [
            TestCaseResult {
              "error": undefined,
              "name": "Is_Even_Number(2)",
              "result": "success",
              "time": 1,
            },
            TestCaseResult {
              "error": {
                "details": "at NUnitLegacyTests.CalculatorTests.Is_Even_Number(Int32 i) in C:\\Users\\james_t\\source\\repos\\NUnitLegacy\\NUnitLegacyTests\\CalculatorTests.cs:line 61
",
                "line": undefined,
                "message": "  Expected: True
  But was:  False
",
                "path": undefined,
              },
              "name": "Is_Even_Number(3)",
              "result": "failed",
              "time": 1,
            },
            TestCaseResult {
              "error": {
                "details": "at MyLibrary.Calculator.Div(Int32 a, Int32 b) in C:\\Users\\james_t\\source\\repos\\NUnitLegacy\\MyLibrary\\Calculator.cs:line 7
at NUnitLegacyTests.CalculatorTests.Exception_In_TargetTest() in C:\\Users\\james_t\\source\\repos\\NUnitLegacy\\NUnitLegacyTests\\CalculatorTests.cs:line 33
",
                "line": undefined,
                "message": "System.DivideByZeroException : Attempted to divide by zero.",
                "path": undefined,
              },
              "name": "Exception_In_TargetTest",
              "result": "failed",
              "time": 26,
            },
            TestCaseResult {
              "error": {
                "details": "at NUnitLegacyTests.CalculatorTests.Exception_In_Test() in C:\\Users\\james_t\\source\\repos\\NUnitLegacy\\NUnitLegacyTests\\CalculatorTests.cs:line 39
",
                "line": undefined,
                "message": "System.Exception : Test",
                "path": undefined,
              },
              "name": "Exception_In_Test",
              "result": "failed",
              "time": 2,
            },
            TestCaseResult {
              "error": {
                "details": "at NUnitLegacyTests.CalculatorTests.Failing_Test() in C:\\Users\\james_t\\source\\repos\\NUnitLegacy\\NUnitLegacyTests\\CalculatorTests.cs:line 27
",
                "line": undefined,
                "message": "  Expected: 3
  But was:  2
",
                "path": undefined,
              },
              "name": "Failing_Test",
              "result": "failed",
              "time": 14,
            },
            TestCaseResult {
              "error": {
                "details": "",
                "line": undefined,
                "message": "couldn't run test for some reason",
                "path": undefined,
              },
              "name": "Inconclusive_Test",
              "result": "skipped",
              "time": 1,
            },
            TestCaseResult {
              "error": undefined,
              "name": "Passing_Test",
              "result": "success",
              "time": 0,
            },
            TestCaseResult {
              "error": undefined,
              "name": "Passing_Test_With_Description",
              "result": "success",
              "time": 0,
            },
            TestCaseResult {
              "error": {
                "details": "",
                "line": undefined,
                "message": "Skipped",
                "path": undefined,
              },
              "name": "Skipped_Test",
              "result": "skipped",
              "time": 0,
            },
            TestCaseResult {
              "error": {
                "details": "",
                "line": undefined,
                "message": "Test exceeded Timeout value of 1ms",
                "path": undefined,
              },
              "name": "Timeout_Test",
              "result": "failed",
              "time": 28,
            },
          ],
        },
      ],
      "name": "NUnitLegacyTests.CalculatorTests",
      "totalTime": undefined,
    },
  ],
  "totalTime": undefined,
}
`;
